tests/utils/url-helpers.test.ts

```diff
@@ -1,25 +1,522 @@
+        'http://www.google.com.whitelisted.com',
+        'http://www.Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ.whitelisted.com',
+        'ja\nva\tscript\r:alert(1)',
+        'java%09script:alert(1)',
+        'java%0ascript:alert(1)',
+        'java%0d%0ascript%0d%0a:alert(0)',
+        'java%0dscript:alert(1)',
+        'Javas%26%2399;ript:alert(1)',
+        '//Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ\twhitelisted.com/',
+        '\u006A\u0061\u0076\u0061\u0073\u0063\u0072\u0069\u0070\u0074\u003aalert(1)',
+        '////whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ//',
+        '////whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/',
+        '///whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ//',
+        '///whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/',
+        '//Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/whitelisted.com',
+        '//Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ@whitelisted.com',
+        '//whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ//',
+        '//whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/',
+        'Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/whitelisted.com',
+        'whitelisted.com;@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ',
+        '////whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2e%2e',
+        '///whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2e%2e',
+        '////whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2e%2e%2f',
+        '///whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2e%2e%2f',
+        '//whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2e%2e%2f',
+        '////whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2f..',
+        '///whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2f..',
+        '//whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2f..',
+        '////whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2f%2e%2e',
+        '///whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2f%2e%2e',
+        '//whitelisted.com@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2f%2e%2e',
+        '/whitelisted.com:80%40google.com',
+        'whitelisted.com@%E2%80%AE@google.com',
+        '////whitelisted.com@google.com//',
+        '////whitelisted.com@google.com/',
+        '///whitelisted.com@google.com//',
+        '///whitelisted.com@google.com/',
+        '//whitelisted.com@google.com//',
+        '//whitelisted.com@google.com/',
+        'whitelisted.com;@google.com',
+        'whitelisted.com.google.com',
+        '////whitelisted.com@google.com/%2e%2e',
+        '///whitelisted.com@google.com/%2e%2e',
+        '////whitelisted.com@google.com/%2e%2e%2f',
+        '///whitelisted.com@google.com/%2e%2e%2f',
+        '//whitelisted.com@google.com/%2e%2e%2f',
+        '////whitelisted.com@google.com/%2f..',
+        '///whitelisted.com@google.com/%2f..',
+        '//whitelisted.com@google.com/%2f..',
+        '////whitelisted.com@google.com/%2f%2e%2e',
+        '///whitelisted.com@google.com/%2f%2e%2e',
+        '//whitelisted.com@google.com/%2f%2e%2e',
+        '//whitelisted.com+&@google.com#+@whitelisted.com/',
+        '//whitelisted.com@https:///Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ/%2e%2e',
+        '//whitelisted.com@https:///google.com/%2e%2e',
+        '//whitelisted.com+&@Ⓛ𝐨𝗰�𝕝ⅆ𝓸ⓜₐℹⓃ｡Ｐⓦ#+@whitelisted.com/'
+      ];
+
+      for (const payload of payloads) {
+        expect(toSafeRedirect(payload, safeBaseUrl) || safeBaseUrl.toString()).toMatch(/^http:\/\/www.example.com\//);
```

Comment:
Shouldn't this be asserting that the outcome of calling `toSafeRedirect` is `undefined`?